File: app/src/main/java/com/sunnyweather/android/ui/place/PlaceViewModel.java
package com.sunnyweather.android.ui.place;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import com.sunnyweather.android.logic.Repository;
import com.sunnyweather.android.logic.model.Place;

import java.util.ArrayList;

public class PlaceViewModel extends ViewModel {
    private MutableLiveData<String> searchLiveData = new MutableLiveData<String>();
    Repository repository = new Repository();
    ArrayList<Place> placeList = new ArrayList<Place>();

    String placeLiveData = Transformations.switchMap(searchLiveData){
        String query ;
        repository.searchPlaces(query);
    }

    public void searchPlaces(String query){
        searchLiveData.setValue(query);
    }

}

File: app/src/main/java/com/sunnyweather/android/ui/place/PlaceViewModel.kt
package com.sunnyweather.android.ui.place

import androidx.lifecycle.*
import androidx.lifecycle.Transformations.switchMap

import com.sunnyweather.android.logic.Repository
import com.sunnyweather.android.logic.dao.PlaceDao
import com.sunnyweather.android.logic.model.Place

class PlaceViewModel : ViewModel() {

    private val searchLiveData = MutableLiveData<String>()

    val placeList = ArrayList<Place>()

    val placeLiveData = switchMap(searchLiveData) { query ->
        Repository.searchPlaces(query)
    }

    fun searchPlaces(query: String) {
        searchLiveData.value = query
    }

    fun savePlace(place: Place)=Repository.savePlace(place)

    fun getSavedPlace()= PlaceDao.getSavedPlace()

    fun isPlaceSaved()= PlaceDao.isPlacesSaved()

}